### PR TITLE
Add docs for Single Sign-On for developers in the Enterprise developer portal

### PR DESCRIPTION
**Pull request by slega**
## Related Issue (if applicable)
https://tyktech.atlassian.net/browse/TT-7646

## Description
- Add docs for SSO in the enterprise developer portal
- Add illustrations to explain the UserGroupMapping

## Motivation and Context
We didn't have any docs for the SSO flow for developers.

## Preview Link
https://deploy-preview-2412--tyk-docs.netlify.app/docs/nightly/tyk-stack/tyk-developer-portal/enterprise-developer-portal/managing-access/enable-sso/

## Types of changes

- [ ] Fixing typo (please merge to production)
- [x] Documenting a new feature (please merge to production)
- [ ] Documentation for future release (please do not merge to production)
- [ ] Something else (please add if needs merging to production or not)

## Checklist

- [x] I have [reviewed the guidelines](../CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](../CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [X] Make sure you have started *your change* off *our latest `master`*.
- [X] I have added a preview link.

### CODE REVIEW

**Comment by rewsmith** [comment]
@slega I've tested the SSO process in a k8s install and it works!  We could however make it easier for k8s users, which frankly is the majority of prospects in EMEA and I presume this is true in other regions also.

### TIB Config/Profiles
For k8s I would suggest we provide example TIB Config and profiles.json.  I did wonder if they better belong in the TIB Getting Started Guide but since they contain Portal specific values, better to create here I think.  Would be nice to include a link to here in the TIB Getting Started Guide

## TIB extraEnvs (not sure if all required but this is what I have)

```
extraEnvs:
  - name: TYK_IB_HTTPSERVEROPTIONS_CERTFILE
    value: "./certs/server.pem"
  - name: TYK_IB_HTTPSERVEROPTIONS_KEYFILE
    value: "./certs/server.key"
  - name: TYK_IB_SSLINSECURESKIPVERIFY
    value: "true"
  - name: TYK_IB_BACKEND_NAME
    value: "in_memory"
  - name: TYK_IB_BACKEND_IDENTITYBACKENDSETTINGS_HOSTS
    value: "redis.tyk-cp:6379"
  - name: TYK_IB_BACKEND_IDENTITYBACKENDSETTINGS_PASSWORD
    value: ""
  - name: TYK_IB_BACKEND_IDENTITYBACKENDSETTINGS_DATABASE
    value: "0"
  - name: TYK_IB_BACKEND_IDENTITYBACKENDSETTINGS_ENABLECLUSTER
    value: "false"
  - name: TYK_IB_BACKEND_IDENTITYBACKENDSETTINGS_MAXIDLE
    value: "1000"
  - name: TYK_IB_BACKEND_IDENTITYBACKENDSETTINGS_MAXACTIVE
    value: "2000"
  - name: TYK_IB_TYKAPISETTINGS_DASHBOARDCONFIG_ENDPOINT
    value: "http://enterprise-portal-svc-tyk-pro.tyk-cp"
  - name: TYK_IB_TYKAPISETTINGS_DASHBOARDCONFIG_PORT
    value: "3001"
  - name: TYK_IB_TYKAPISETTINGS_DASHBOARDCONFIG_ADMINSECRET
    value: "123456"
```

## Example TIB profiles.json ConfigMap (Contains both Admin and Developer profiles)
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: tyk-tib-profiles-conf
data:
  profiles.json: |
    [{
      "ActionType": "GenerateOrLoginUserProfile",
      "ID": "{ID of your TIB profile}",
      "OrgID": "0",
      "IdentityHandlerConfig": {
        "DashboardCredential": "{portal API secret}"
      },
      "ProviderConfig": {
        "CallbackBaseURL": "http://{TIB host}:{TIB port}",
        "FailureRedirect": "http://{portal host}:{portal port}/?fail=true",
        "UseProviders": [
          {
            "Name": "openid-connect",
            "Key": "{oAuth2.0 key}",
            "Secret": "{oAuth2.0 secret}",
            "DiscoverURL": "OIDC well-known endpoint"
          }
        ]
      },
      "ProviderName": "SocialProvider",
      "ReturnURL": "http://{portal host}:{portal port}/sso",
      "Type": "redirect"
    },
    {
      "ActionType": "GenerateOrLoginDeveloperProfile",
      "ID": "{ID of your TIB profile}",
      "OrgID": "0",
      "IdentityHandlerConfig": {
        "DashboardCredential": "{portal API secret}"
      },
      "ProviderConfig": {
        "CallbackBaseURL": "http://{TIB host}:{TIB port}",
        "FailureRedirect": "http://{portal host}:{portal port}/?fail=true",
        "UseProviders": [
          {
            "Name": "openid-connect",
            "Key": "{oAuth2.0 key}",
            "Secret": "{oAuth2.0 secret}",
            "DiscoverURL": "OIDC well-known endpoint"
          }
        ]
      },
      "ProviderName": "SocialProvider",
      "ReturnURL": "http://{portal host}:{portal port}/sso",
      "Type": "redirect",
      "DefaultUserGroupID": "1"
    }]
```

### Enterprise Portal Env Vars needed for SSO
```
extraEnvs:
  - name: PORTAL_API_SECRET
    value: "123456"
```

Maybe need a note on where to find `PORTAL_API_SECRET`.
Needs to be documented as an env var and needs to match the equivalent Env Var in TIB (`TykAPISettings.DashboardConfig.AdminSecret`).  Defaults to empty otherwise.

### Login Form
I didn't use this but the form method action should end with /openid-connect.  Also, shouldn't submit be a button?
To kick off the login I just copy a TIB Profile URL into a browser, e.g. `http://localhost:3010/auth/portal-admin-sso/openid-connect`

### SSO profile for Developers

IdentityHandlerConfig.DashboardCredential should be `{portal API secret}` not `{ID of your TIB profile}`

```
"IdentityHandlerConfig": {
    "DashboardCredential": "{portal API secret}"
 }
```
 
I did not test User Group Mapping but it seems clear from the docs to me.  It would be helpful I think if we logged (debug logs) the JWT from the IdP.  We do that in the Dashboard and it helps when you need to see what claims are coming through. (Great if we do this already, I didn't check)

**Review by rewsmith — APPROVED** [review]
LGTM!

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/managing-access/enable-sso.md [hunk 1]
```diff
@@ -17,78 +17,434 @@ If you are interested in getting access, contact us at [support@tyk.io](<mailto:
...
+However, for developers there is one additional.
+
+#### User group mapping
+In order to land a developer into the right API Consumer organisation, it is necessary to configure the UserGroupMapping
+in the TIB profile that creates a binding between user groups in your IDP and developer teams in the portal.
+
+{{< img src="/img/dashboard/portal-management/enterprise-portal/user-group-mapping.png" alt="User group mapping" width="600">}}
+
+To define the user group mapping for your developer audience, you need to add the UserGroupMapping object to the corresponding TIB profile:
+```json
```
Comment (letzya) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/managing-access/enable-sso.md]:
```suggestion
```yaml
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/managing-access/enable-sso.md [hunk 2]
```diff
@@ -17,78 +17,434 @@ If you are interested in getting access, contact us at [support@tyk.io](<mailto:
...
+
+{{< note info >}}
+**Nuances of OIDC configuration**
+
+To ensure that the portal can log in a user with your OIDC Identity provider, you may need to either explicitly specify the email scopes in a profile
+configuration or configure your IDP to include the email claim in the JWT. Failure to include the email scope in the JWT
+would result in the portal not having access to the user's email.
+
+As an example, for Okta, you can use the following configuration:
+```json
```
Comment (letzya) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/managing-access/enable-sso.md]:
```suggestion
```yaml
```

**Pull request merged**